core/base/marchingTetrahedra/MarchingTetrahedra.h

```diff
@@ -0,0 +1,4047 @@
+/// \ingroup base
+/// \class ttk::MarchingTetrahedra
+/// \author Robin G. C. Maack <maack@rptu.de>
+/// \date May 2023.
+///
+/// \brief TTK processing package for Marching Tetra/Triangles computations
+///
+/// Given an input scalar field with labels attached to the point data this
+/// class executes the marching tetrahedra/triangles algorithm. It has three
+/// options that either separate each label with a single separating geometry
+/// inbetween two labels, or a separating geometry enclosing each label
+/// (detailed and fast mode).
+///
+/// \b Related \b publication \n
+/// "Parallel Computation of Piecewise Linear Morse-Smale Segmentations" \n
+/// Robin G. C. Maack, Jonas Lukasczyk, Julien Tierny, Hans Hagen,
+/// Ross Maciejewski, Christoph Garth \n
+/// IEEE Transactions on Visualization and Computer Graphics \n
+///
+/// \sa ttkMarchingTetrahedra.cpp %for a usage example.
+///
+/// \b Online \b examples: \n
+///   - <a
+///   href="https://topology-tool-kit.github.io/examples/morseSmaleSegmentation_at/">Morse-Smale
+///   segmentation example</a> \n
+
+#pragma once
+
+// base code includes
+#include <Triangulation.h>
+
+#include <queue>
+
+/**
+ * Lookup table to retrieve the indices of the edges (0-5) and triangles (6-9)
+ * of the tetrahedron used for each generated triangle. The indices are ordered
+ * by the combination of vertex indices that span the simplex
+ * (e.g. [0,1] = edge 0, [0,2] = edge 1, [1,2] = edge 3, [0,1,3] = triangle 7).
+ * Every 3 entries represent one generated triangle by using the middle of the
+ * edges and triangles respectively.
+ * The comments next to each line first indicate the number of unique vertex
+ * labels on the tetrahedron, the 2nd label, 3rd label, 4th label, and its index
+ * (the first label is alway considered to be 0).
+ * This allows to directly retrieve the triangulation from this lookup table by
+ * supplying a bit code, where the 4th label supplies the two LSB
+ */
+constexpr int tetLookupWall[28][15] = {
```

Comment:
These kind of data files are often added in `.inl`  files. This would keep data / implemetation asides